app/master/build.py

```diff
@@ -451,6 +452,21 @@ def _create_build_artifact(self):
         self._build_artifact.write_timing_data(self._timing_file_path, self._read_subjob_timings_from_results())
         self._artifacts_archive_file = app.util.fs.compress_directory(self._build_results_dir(), 'results.tar.gz')
 
+    def _delete_temporary_build_artifact_files(self):
+        """
+        Delete the temporary build result files that are no longer needed, due to the creation of the
+        build artifact tarball (results.tar.gz).
+
+        ONLY call this method after _create_build_artifact() has completed. Otherwise we have lost the build results.
+        """
+        build_result_dir = self._build_results_dir()
+        for path in os.listdir(build_result_dir):
+            # The build result tar-ball (results.tar.gz) is also stored in this same directory, so we must
+            # skip deleting it.
+            if path == 'results.tar.gz':
```

Comment:
Should we make this a constant somewhere? Not like we'll change it but it might make it slightly easier to track references.